frontend/src/components/CloseSuspendReasonModal.js

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Form, FormGroup, ErrorMessage, Label, Fieldset, Radio, Textarea,
+} from '@trussworks/react-uswds';
+import Modal from './Modal';
+import { GOAL_CLOSE_REASONS, GOAL_SUSPEND_REASONS } from '../Constants';
+import './CloseSuspendReasonModal.css';
+
+const CloseSuspendReasonModal = ({
+  modalRef, goalId, newStatus, onSubmit, resetValues,
+}) => {
+  const [closeSuspendReason, setCloseSuspendReason] = useState('');
+  const [closeSuspendContext, setCloseSuspendContext] = useState('');
+  const [showValidationError, setShowValidationError] = useState(false);
+
+  useEffect(() => {
+    // Every time we show the modal reset the form.
+    setCloseSuspendReason('');
+    setCloseSuspendContext('');
+    setShowValidationError(false);
+  }, [resetValues]);
+
+  const reasonDisplayStatus = newStatus === 'Completed' ? 'closing' : 'suspending';
+  const reasonRadioOptions = newStatus === 'Completed' ? GOAL_CLOSE_REASONS : GOAL_SUSPEND_REASONS;
+  const ReasonChanged = (e) => {
+    setCloseSuspendReason(e.target.value);
+    setShowValidationError(false);
+  };
+  const generateReasonRadioButtons = () => reasonRadioOptions.map((r) => (
+    <Radio
+      id={`${goalId}-${r}-radio-reason`}
+      key={`${goalId}-${r}-radio-reason`}
```

Comment:
HTML ids must start with a letter, and they can't have spaces, please adjust line 33 accordingly